### PR TITLE
added native implementation for sum of vector in linalg

### PR DESCRIPTION
**Pull request by curiousguy13**
addresses issue #2667 .

### CODE REVIEW

File: src/shogun/mathematics/linalg/internal/implementation/VectorSum.h [hunk 1]
```diff
@@ -76,6 +76,42 @@ struct vector_sum
...
+	 * @param vec a vector whose sum has to be computed
+	 * @param len the length of the vector
+	 * @return the vector sum \f$\sum_i a_i\f$
+	 */
+	static T compute(T* vec, index_t len)
+	{
+		T result=static_cast<T>(0);
+		for(index_t i=0; i<len; ++i)
+			result+=vec[i];
+		return result;
```
Comment (lambday) [inline src/shogun/mathematics/linalg/internal/implementation/VectorSum.h]: we can use std::accumulate here :)

**Comment by curiousguy13** [comment]
@lambday is there anything else that i need to change here?

**Comment by lambday** [comment]
Good to go! Merging :)

File: src/shogun/mathematics/linalg/internal/implementation/VectorSum.h [hunk 2]
```diff
@@ -76,6 +76,40 @@ struct vector_sum
...
+	/**
+	 * Method that computes the sum of SGVectors
+	 * 
+	 * @param vec a vector whose sum has to be computed
+	 * @param len the length of the vector
+	 * @return the vector sum \f$\sum_i a_i\f$
+	 */
+	static T compute(T* vec, index_t len)
+	{
+		return std::accumulate(vec,vec+len,0);
```
Comment (lisitsyn) [inline src/shogun/mathematics/linalg/internal/implementation/VectorSum.h]:
```
In file included from /Users/lisitsyn/Matters/OSS/shogun/src/shogun/kernel/CustomKernel.cpp:23:
In file included from /Users/lisitsyn/Matters/OSS/shogun/src/shogun/mathematics/linalg/linalg.h:188:
In file included from /Users/lisitsyn/Matters/OSS/shogun/src/shogun/mathematics/linalg/internal/modules/Redux.h:37:
/Users/lisitsyn/Matters/OSS/shogun/src/shogun/mathematics/linalg/internal/implementation/VectorSum.h:111:15: error: no member named 'accumulate' in namespace 'std'
                return std::accumulate(vec,vec+len,0);
```

Comment (lambday) [inline src/shogun/mathematics/linalg/internal/implementation/VectorSum.h]: Errr. my bad! Shouldn't have merged.
@curiousguy13 the header??

Comment (curiousguy13) [inline src/shogun/mathematics/linalg/internal/implementation/VectorSum.h]: Oh, sorry about that!
@lambday should i send a separate PR for this ?

Comment (lisitsyn) [inline src/shogun/mathematics/linalg/internal/implementation/VectorSum.h]: Guys no worries I'll commit this directly, `#include <numeric>` solves the problem

Comment (curiousguy13) [inline src/shogun/mathematics/linalg/internal/implementation/VectorSum.h]: @lisitsyn Thanks!
